### PR TITLE
Add get_default_keybindings script

### PR DESCRIPTION
**Pull request by tshino**
Hi, this will be the first step of adding automated workflow #10.

This PR adds `scripts/get_default_keybindings/` directory.
This should be able to run by following commands on any one of the three OSes, Windows, Linux, and macOS.
```
cd scripts/get_default_keybindings
npm install
npm start
```
It outputs a file `scripts/{platform}.keybindings.raw.json`.

In addition to the raw Default Keybindings JSON, I inserted a header line to the output file.
```
// Default Keybindings of Visual Studio Code 1.66.2 for Windows
```
This is for the traceability of the file.
I found that `process_json.py` may need some change for this.

Would you please confirm that this works expectedly?
The next PR will be adding a YAML file to `.github/` to create a new workflow that allows running this script on GitHub Actions.

### CODE REVIEW

**Comment by BenF-lively** [comment]
Great work!

I don't have time today to try the script, but it's exciting to see the contribution. I don't mind making the corresponding changes to `process_json.py`. (I'm hoping we can get a relatively clean diff in the output files, should be possible!)

**Comment by moltenform** [comment]
Great work!

I don't have time today to try the script, but it's exciting to see the contribution. I don't mind making the corresponding changes to `process_json.py`. (I'm hoping we can get a relatively clean diff in the output files, should be possible!)

**Comment by codebling** [comment]
@moltenform do you think it will simplify the process if we convert the Python script to Node? I don't mind doing this if it does. I'm thinking that it will be easier to integrate and leverage tshino's code if we do that. 

Maybe it's also a good time to ask you about the finishing touches. It looks like we:
1) adjust line spacing
2) fix line terminators/endings on Windows
3) remove a comment line
4) remove other available commands

Why do we do 1) and 2)? Don't those get preserved if they are already in the .raw file? 
3) and 4) make the file standard JSON besides getting rid of some unnecessary data, if I understand correctly. 

I'll probably test and merge tshino's change and set them a contributor for the repo, if that's ok by you.

**Comment by codebling** [comment]
> Would you please confirm that this works expectedly?
Forgot to mention that I tested this yesterday and it does work as expected.

**Comment by tshino** [comment]
Thanks for testing the script!
Initially, I planned to make separate PRs, but I decided to keep using this PR for further commits.
I will add the next commit to this PR soon.

**Comment by tshino** [comment]
I have just added a YAML file into `.github/workflows`.

As far as I know, GitHub Actions can be triggered only if the workflow file is present in the default branch (master).
So, in order to test the workflow, we need to merge this PR first.
So I didn't test this workflow yet.
Though it is basically the same as the one for my vscode extension repository, it has some changes.

The workflow runs only when we manually start it.
```
on:
  # schedule:
  #   - cron: '22 11 * * 5' # weekly
  workflow_dispatch:
```
After a successful run and if everything is fine, you can comment out the `schedule` section.

So, if this PR looks good, please merge it to the master branch.
Then we can test the workflow actually.
If it runs perfectly, a new PR for updating the JSON files (vscode 1.66.2) will appear on this repository.

**Review by codebling — APPROVED** [review]

**Comment by codebling** [comment]
Looks great! I believe that, using GitHub Actions, one can trigger workflows when there are changes in the upstream repo, e.g. a new release of VS Code. I think this might be a better way to do this than a cron job. I can look into the trigger. 

Otherwise just waiting on moltenform's feedback before merging.

**Comment by codebling** [comment]
>  So, in order to test the workflow, we need to merge this PR first.

GitLab offers a CLI for running/testing automation, but I'm not aware of anything like this for GitHub. I'll have another look just in case this has changed.

File: scripts/get_default_keybindings/main.js
Comment (BenF-lively) [inline scripts/get_default_keybindings/main.js:9]: In case I missed someplace where it's already been done, consider deleting the contents and/or assert that it's empty. Just in the unlikely case an artifact from another version affects anything!

**Review by BenF-lively — APPROVED** [review]

File: scripts/get_default_keybindings/main.js
Comment (codebling) [reply scripts/get_default_keybindings/main.js:9]: Agreed, I think a temp path would work better here, I might make this change

**Comment by codebling** [comment]
I merged and ran the Workflow. Looks like there are a few minor issues. I'll have a look at them a bit later

File: scripts/get_default_keybindings/main.js
Comment (tshino) [reply scripts/get_default_keybindings/main.js:9]: Good catch! Thanks.

**Comment by tshino** [comment]
Thank you for merging and inviting me to the collaborators for this repo!

**Comment by tshino** [comment]
I think adding this trigger helps us to debug the workflow on a PR.
```
on:
  pull_request:
    branches:
      - master
    paths:
      - '.github/workflows/*'
      - 'scripts/**'
```
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Comment by tshino** [comment]
Shall I continue debugging?

**Comment by codebling** [comment]
I am still planning on adding the code change you listed above and on fixing CI. If you'd like to keep continue debugging before then, feel free, otherwise I will take care of it. I did spend some time looking into triggers but it looked like I was wrong (I should have guessed - I am not that familiar with GitHub Actions, so the fact that you were using cron should have already told me I was wrong).

**Comment by codebling** [comment]
Opened #12. More changes to come but those will have to wait for a bit.

**Comment by codebling** [comment]
I restored the cron and manual triggers for the workflow that I accidentally wiped out in #12. The workflow just ran and opened #14. 

The workflow output has [a comment at the top of the file with the version number](https://github.com/codebling/vs-code-default-keybindings/pull/14/files#diff-4d5744c3bc68111069a5caef51b34620529ed14f14736aa08ee17bad7d01451dR1), whereas normally the scripts strip all of the comments (which are technically not allowed in JSON).  @moltenform @BenF-lively is this ok? VS Code can definitely handle the comments but it seemed like you wanted them all removed from the file.

**Comment by codebling** [comment]
Just want to say again thanks @tshino ! This is working great. 

I notice the Mac OS job does seem to fail often with `Error: read ECONNRESET`. I'm not sure this is something we can solve, but if you have any ideas about what might be happening, let me know,. I think ECONNRESET can only stem from real networking issues so I don't know how we could change anything, or why it only happens with the Mac OS job. It happens often enough that I just ignore the notifications that the job failed, which is fine for me because it runs quite often and usually fixes itself soon enough that new PR would be opened the same day as a new version was released.

Overall, really great work! Thank you!

**Comment by codebling** [comment]
@tshino thanks for the idea! It's the same package (renamed) but maybe a newer version would help.

**Pull request merged**
